src/main/kotlin/io/wusa/formatter/SemanticVersionFormatter.kt

```diff
@@ -65,11 +65,12 @@ class SemanticVersionFormatter {
 
         private fun hasVersionPrerelease(info: Info) = info.version.prerelease != ""
 
-        private fun appendSuffix(version: String, snapshotSuffix: String): String {
-            if (snapshotSuffix != "") {
-                return "$version-$snapshotSuffix"
+        private fun appendSuffix(version: String, branches: Branches, info: Info): String {
+            val regexFormatterPair = RegexResolver.findMatchingRegex(branches, info.branch.name)
+            regexFormatterPair?.let {
+                return "$version-${regexFormatterPair.snapshotSuffix}"
             }
-            return "$version"
+            return "$version-${SemverGitPluginExtension.DEFAULT_SNAPSHOT_SUFFIX}"
```

Comment:
Nice, THX! LGTM. Imho it's still nice to keep the snapshotSuffix != "" check enabled, at least for DEFAULT_SNAPSHOT_SUFFIX.

Reply:
Oh you are right I changed it.